### PR TITLE
PAT-989 increased hit limit

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt [hunk 1]
```diff
@@ -50,6 +51,7 @@ class PrisonerSearchService(
     return query?.let {
       val searchSourceBuilder = SearchSourceBuilder().apply {
         query(query.withDefaults(searchCriteria))
+        size(RESULT_HITS_MAX)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:54]: @Mjwillis This seems reasonable to me for all searches but did you have any intent for the general searches to bring back only 10 offenders?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            .withHeader("Total-Records", prisonerNumbers.size.toString())
+            .withBody(gson.toJson( prisonerNumbers.map { IDs(it) }))))
+        prisonerNumbers.forEach {
+          prisonMockServer.stubFor(get(urlEqualTo("/api/offenders/$it"))
+          .willReturn(aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withBody(getOffenderBookingJson(it))))
+        }
+
+      setupIndexes()
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt:39]: Very minor, looks like formatting has gone a bit weird, I think we settled of 4 tabs size for Kotlin

Comment (lbennett-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt:39]: thanks - will set that up

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt [hunk 3]
```diff
@@ -21,6 +21,7 @@ class PrisonerSearchService(
 ) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
+    const val RESULT_HITS_MAX = 1000
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:24]: Wonder if this at some point should be config to allow for tests to tweak it etc

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:54]: @andymarke no it was a missing bit on my part.  I notice that UoF only returns 10 results which is not strictly correct!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt [hunk 4]
```diff
@@ -61,6 +63,7 @@ class PrisonerSearchService(
     val query = queryBuilder(searchCriteria)
     return query?.let {
       val searchSourceBuilder = SearchSourceBuilder().apply {
+        size(RESULT_HITS_MAX)
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:66]: In the two `queryBy` methods it looks like there's a few lines of duplicated code (up to building the `searchRequest`). Could we extract out the duplication?

Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:66]: There is a difference in the lines of code so I didn't think it was worth it

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt (hunk 3 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerSearchService.kt:24]: yeah - wondered about that - but it took a long time to get the int tests working. Hopefully I'll get chance to do that

**Pull request merged**